contracts/SnappAuction.sol

```diff
@@ -56,9 +75,76 @@ contract SnappAuction is SnappBase {
         return auctions[slot].appliedAccountStateIndex != 0;
     }
 
+    function getStandingOrderHash(uint16 userId, uint128 nonce) public view returns (bytes32) {
+        return reservedAccountOrders[userId][nonce].orderHash;
+    }
+    
+    function getStandingOrdervalidFrom(uint16 userId, uint128 nonce) public view returns (uint) {
+        return reservedAccountOrders[userId][nonce].validFrom;
+    }
+
+    function getStandingOrderValidityTo(uint16 userId, uint128 nonce) public view returns (uint) {
+        return reservedAccountOrders[userId][nonce].validTo;
+    }
+
     /**
      * Auction Functionality
      */
+    function placeStandingSellOrder(
+        uint8[] memory buyToken,
+        uint8[] memory sellToken,
+        uint128[] memory buyAmount,
+        uint128[] memory sellAmount
+    ) public onlyRegistered() {
+        
+        // Update Auction Hash based on request
+        uint16 accountId = publicKeyToAccountMap(msg.sender);
+        require(accountId <= AUCTION_RESERVED_ACCOUNTS, "Accout is not a rented account");
+
+        bytes32 orderHash;
+        uint nrOrder = buyToken.length;
+        require(nrOrder <= AUCTION_RESERVED_ACCOUNT_BATCH_SIZE, "Too many orders");
+        
+        if (
+            auctionIndex == MAX_UINT ||
```

Comment:
If solidity is a "short circuiting" I would think that we could be more efficient if the auctionIndex check occurred second. I really don't like this checking the contract initialization case every single time.